sidecar/app/query/base.py

```diff
@@ -201,8 +201,11 @@ def run_query(self, query: Query):
             )
 
         self.running_queries[query.query_id] = query
-        query.start()
-        del self.running_queries[query.query_id]
+        try:
+            query.start()
+        finally:
+            # always remove this
+            del self.running_queries[query.query_id]
```

Comment:
is there a way to test this behavior to make sure we don't miss this edge case by accidentally removing this code?

Reply:
added!